ledger-core/v2/virtual/object/orderedPending.go

```diff
@@ -0,0 +1,37 @@
+package object
+
+import (
+	"github.com/insolar/assured-ledger/ledger-core/v2/conveyor/smachine"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/injector"
+)
+
+type AwaitOrderedPendingSM struct {
+	smachine.StateMachineDeclTemplate
+	sync smachine.SyncLink
+	stop smachine.BargeIn
+}
+
+func (sm *AwaitOrderedPendingSM) InjectDependencies(_ smachine.StateMachine, _ smachine.SlotLink, _ *injector.DependencyInjector) {
+	// No-op.
+}
+
+func (sm *AwaitOrderedPendingSM) GetInitStateFor(smachine.StateMachine) smachine.InitFunc {
+	return sm.Init
+}
+
+func (sm *AwaitOrderedPendingSM) GetStateMachineDeclaration() smachine.StateMachineDeclaration {
+	return sm
+}
+
+func (sm *AwaitOrderedPendingSM) Init(ctx smachine.InitializationContext) smachine.StateUpdate {
+	if !ctx.Acquire(sm.sync) {
+		panic("failed to acquire semaphore")
+	}
+
+	// Sync will be released on machine stop.
+	sm.stop = ctx.NewBargeIn().WithStop()
+
+	return ctx.Jump(func(ctx smachine.ExecutionContext) smachine.StateUpdate {
+		return ctx.Sleep().ThenRepeat()
```

Comment:
this looks wrong. if SM would be woken up for any reason then it will either panic with failure to acquire lock or acquires twice or may be MutableExecute checks double acquire, it's not clear. Create a separate step for spin sleep.

Reply:
@ruz Fixed